### ISSUE TITLE
False positive: too-many-function-args for numpy

### ISSUE THREAD

**Issue by jwkvam** [PyCQA/pylint#2326]
### Steps to reproduce
`test.py`
```
import numpy as np
x = np.fromfile('hello.npy')
x.reshape((1, 5))
```
https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.reshape.html#numpy.ndarray.reshape
```
pylint test.py
test.py:3:6: E1121: Too many positional arguments for method call (too-many-function-args)
```

### Current behavior
too-many-function-args

### Expected behavior
No error

### pylint --version output
pylint 2.0.0
astroid 2.0.1
Python 3.6.5 | packaged by conda-forge | (default, Apr  6 2018, 13:44:09)
[GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)]

**Comment by PCManticore** [comment]
Hi @jwkvam 

What version of `numpy` do you have? Locally using the same pylint and Python version, I'm not getting the error, getting instead:
```
************* Module a
a.py:1:0: C0111: Missing module docstring (missing-docstring)
a.py:2:0: C0103: Constant name "x" doesn't conform to UPPER_CASE naming style (invalid-name)
```

**Comment by jwkvam** [comment]
After some further debugging it seems this is only triggered with:
```
extension-pkg-whitelist=numpy
```
in my `pylintrc`. I've tried this with numpy 1.14.5 and 1.15.0. I forget why I whitelisted numpy in the first place, it was likely years ago. I don't know if this is still a bug :/

**Comment by PCManticore** [comment]
Thanks, now I can also reproduce it with `extension-pkg-whitelist`.

**Comment by NotSqrt** [comment]
Hi,

Same kind of behaviour with :
```python
import numpy as np
np.timedelta64(10, 'ms')
```

I also have `extension-pkg-whitelist=numpy` in my pylintrc, due to all the `no-member` errors that pylint generated at the time for numpy (I haven't checked if these errors are still there).

Thanks !

**Comment by Pluckerpluck** [comment]
I can confirm that using:

```python
one_day = np.timedelta64(1, 'D')
```

results in `E1121: Too many positional arguments for constructor call (too-many-function-args)`

I have no `.pylintrc` and have the following versions:

```
pylint 2.2.2
astroid 2.1.0
Python 3.6.6
numpy 1.15.4
```

**Comment by polosatik** [comment]
Same here: np.timedelta64(1, 'D)
pylint 2.2.2
astroid 2.1.0
Python 3.6.1
numpy 1.15.1

It might be something similar to what causes PyCharm to misinterpret skeletons for numpy.
https://youtrack.jetbrains.com/issue/PY-23521

**Comment by krispharper** [comment]
I'm also seeing this issue with `np.timedelta64(1, "D")` and

```
astroid   2.2.2
numpy     1.15.1
pylint    2.3.0
python    3.6.8
```

**Comment by Mazata** [comment]
Same problem for me with np.timedelta64(1,'D'): "too many positional arguments"

astroid    2.2.5
numpy    1.16.2
pylint      2.3.1
python    3.7.2

**Comment by waiyip-aquabyte** [comment]
I still see this in 
pylint 2.4.4                                                                                                                                                                                                                            astroid 2.3.3                                                                                                                                                                                                                           Python 3.7.3 (default, Dec 29 2019, 20:14:33)                                                                                                                                                                                           [GCC 9.2.1 20191130]                

>>> numpy.version.version                                                                                                                                                                                                               '1.18.2'

**Comment by EdmundsEcho** [comment]
This is a helpful tool.  Thank you.

FYI - I'm getting this error with the following versions:
```
pylint 2.5.3
astroid 2.4.2
Python 3.7.3 (default, Jun  2 2020, 13:01:47)
[Clang 11.0.3 (clang-1103.0.32.59)]

numpy: 1.15.0
```

\- E

**Comment by hippo91** [comment]
@EdmundsEcho @waiyip-aquabyte i'm unable to reproduce the bug with an astroid version >= 2.3.3.

**Comment by SagaraBattousai** [comment]
I'm also having this issue with numpy reshape (I think its because I am reshaping to add dimension so I'll use np.newaxis for now but it looks messier for this particular line of code.

Using :
   pylint==2.8.3
   numpy==1.18.5

and getting:  E1121: Too many positional arguments for method call (too-many-function-args) as a response.

**Comment by ntraut** [comment]
We can get the error with:
```
x = np.zeros(12)
x.reshape(3, 2, 2)
```
It is because pylint doesn't understand that we can pass each element of the shape parameter as a separated argument, as stated in the [documentation.](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.reshape.html#)

**Comment by ntraut** [comment]
> @ntraut could you open a new bug issue please ? This one is unlikely to get any attention because it's closed.

i created the new issue #7883
